regcomp.c

```diff
@@ -23013,23 +23013,54 @@ S_get_extended_utf8_msg(pTHX_ const UV cp)
 #endif /* end of ! PERL_IN_XSUB_RE */
 
 STATIC REGEXP *
-S_compile_wildcard(pTHX_ const char * name, const STRLEN len,
+S_compile_wildcard(pTHX_ const char * subpattern, const STRLEN len,
                          const bool ignore_case)
 {
+    /* Pretends that the input subpattern is qr/subpattern/aam, compiling it
+     * possibly with /i if the 'ignore_case' parameter is true.  Sets up the
+     * debugging info */
+
     U32 flags = PMf_MULTILINE|PMf_WILDCARD;
     REGEXP * subpattern_re;
 
+#  ifdef DEBUGGING
+    bool turned_off_PL_debug = FALSE;
+    bool turned_off_RE_DEBUG_FLAGS = FALSE;
+    GET_RE_DEBUG_FLAGS_DECL;
+#  endif
+
     PERL_ARGS_ASSERT_COMPILE_WILDCARD;
 
     if (ignore_case) {
         flags |= PMf_FOLD;
     }
     set_regex_charset(&flags, REGEX_ASCII_MORE_RESTRICTED_CHARSET);
 
-    subpattern_re = re_op_compile_wrapper(sv_2mortal(newSVpvn(name, len)),
+#  ifdef DEBUGGING
+    if (! isDEBUG_WILDCARD) {
+        if (DEBUG_r_TEST) {
+            PL_debug &= ~ DEBUG_r_FLAG;
+            turned_off_PL_debug = TRUE;
+        }
+        else if (re_debug_flags) {
+            turned_off_RE_DEBUG_FLAGS = TRUE;
+            sv_setuv_mg(get_sv(RE_DEBUG_FLAGS, GV_ADD), 0);
+        }
+    }
+#  endif
+
+    subpattern_re = re_op_compile_wrapper(sv_2mortal(newSVpvn(subpattern, len)),
                                         /* Like in op.c, we copy the compile
                                          * time pm flags to the rx ones */
                                         (flags & RXf_PMf_COMPILETIME), flags);
+#  ifdef DEBUGGING
+    if (turned_off_PL_debug) {
```

Comment:
Is this "undo the changes we made" pattern safe when the call above dies? Do we have a test for that?